pghoard/restore.py

```diff
@@ -432,9 +484,40 @@ def download_progress(current_pos, expected_max, end=""):
         print("On systemd based systems you can run systemctl start postgresql")
         print("On SYSV Init based systems you can run /etc/init.d/postgresql start")
 
+    def download_one_backup(self, *, basebackup_data_file, progress_callback, site):
+        dl_dir = os.path.join(self.config["backup_location"], self.config["path_prefix"], site, "basebackup_incoming")
+        compat.makedirs(dl_dir, exist_ok=True)
+        try:
+            tmp = tempfile.NamedTemporaryFile(dir=dl_dir, prefix="basebackup.", suffix=".pghoard")
+            metadata = self.storage.get_basebackup_file_to_fileobj(
+                basebackup=basebackup_data_file,
+                fileobj=tmp,
+                progress_callback=progress_callback)
+            progress_callback(1, 1)
+            self.log.info("Downloaded %r", basebackup_data_file)
+            tmp.seek(0)
+        except:  # pylint: disable=bare-except
+            self.log.exception("Problem downloading a backup file: %r", basebackup_data_file)
+            os.unlink(tmp)
```

Comment:
`tmp` is a `NamedTemporaryFile` object, not a filename.  Use `tmp.close()` here instead and move the `tmp = tempfile...` call before the try block to avoid a `NameError` here in case creating the temporary file fails for some reaosn.